scripts/salt-setup.sh

```diff
@@ -46,6 +46,37 @@ function apply_optional_states {
   fi
 }
 
+# This adds entities to the 'roles' key to be used during actual cluster install. Hence, this does not use the temporary config location.
+function add_single_role_for_cluster_salt {
+  local role=${1}
+  ${SALT_PATH}/bin/salt-call --local grains.append roles ${role} -l info --log-file=/tmp/salt-build-${saltenv}.log --log-file-level=debug
+}
+
+function add_prewarmed_roles {
+  if [ "${INCLUDE_FLUENT}" == "Yes" ]; then
+    # Note: This will need to be changed if making changes to versions etc in the prewarmed image.
+    local fluent_prewarmed=${FLUENT_PREWARM_TAG}
+    echo "Adding ${fluent_prewarmed} to the list of roles for the final image"
+    add_single_role_for_cluster_salt ${fluent_prewarmed}
+  fi
+
+  if  [ "${STACK_TYPE}" == "CDH" -a ! -z "${CLUSTERMANAGER_VERSION}" -a ! -z "${CLUSTERMANAGER_BASEURL}" -a ! -z "${CLUSTERMANAGER_GPGKEY}" -a ! -z "${STACK_VERSION}" -a ! -z "${STACK_BASEURL}" -a ! -z "${STACK_REPOID}" ]; then
+    local prewarmed="prewarmed_v1"
```

Comment:
if the fluentd variable is extracted maybe this should be also

Reply:
Makes sense. Will make the change if the patches are going in.